cooked-validators/src/Cooked/Tx/Constraints/Type.hs

```diff
@@ -238,6 +251,12 @@ instance Show RawModTx where
   show Id = "Id"
   show (RawModTx _) = "RawModTx"
 
+data Collateral
+  = CollateralAuto
+  | CollateralNone
```

Comment:
Not so obvious when this ctor is useful.

Reply:
I imagine you might want to disable collaterals in the tx options when you intend to tweak the tx manually (txMod) afterwards. Is it equivalent to `CollateralUtxos []` though?

Reply:
Good eye, thanks @florentc, it is equivalent to `CollateralUtxos []`, I removed `CollateralNone`